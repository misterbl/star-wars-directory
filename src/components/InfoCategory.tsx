import React from "react";

interface IInfoCategory {
  list: string[];
  title: string;
}
const InfoCategory = ({ title, list }: IInfoCategory) => (
  <div className="info-category d-flex flex-column">
    <div>{title}</div>
    {list.map(element => (
      <p key={element}>{element}</p>
    ))}
  </div>
);

export default InfoCategory;
