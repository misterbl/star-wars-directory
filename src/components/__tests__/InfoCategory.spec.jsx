import * as React from "react";
import { shallow } from "enzyme";
import InfoCategory from "../InfoCategory";

describe("InfoCategory", () => {
  const props = {
    title: "title",
    list: ["item1", "item2"]
  };
  const wrapper = shallow(<InfoCategory {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
