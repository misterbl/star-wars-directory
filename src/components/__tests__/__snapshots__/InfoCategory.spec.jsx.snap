// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCategory matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InfoCategory
    list={
      Array [
        "item1",
        "item2",
      ]
    }
    title="title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          title
        </div>,
        Array [
          <p>
            item1
          </p>,
          <p>
            item2
          </p>,
        ],
      ],
      "className": "info-category d-flex flex-column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "title",
        },
        "ref": null,
        "rendered": "title",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "item1",
        "nodeType": "host",
        "props": Object {
          "children": "item1",
        },
        "ref": null,
        "rendered": "item1",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "item2",
        "nodeType": "host",
        "props": Object {
          "children": "item2",
        },
        "ref": null,
        "rendered": "item2",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            title
          </div>,
          Array [
            <p>
              item1
            </p>,
            <p>
              item2
            </p>,
          ],
        ],
        "className": "info-category d-flex flex-column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "title",
          },
          "ref": null,
          "rendered": "title",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "item1",
          "nodeType": "host",
          "props": Object {
            "children": "item1",
          },
          "ref": null,
          "rendered": "item1",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "item2",
          "nodeType": "host",
          "props": Object {
            "children": "item2",
          },
          "ref": null,
          "rendered": "item2",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
